controllers/api/userRoutes.js

```diff
@@ -1,9 +1,37 @@
 const router = require('express').Router();
+const { User } = require('../../models');
 
-// Login
 router.post('/login', async (req, res) => {
-    // TODO: Fill in
-    res.send(200).json({ message: "Successful login" });
-});
+    try {
+        const userData = await User.findOne({
+            where: {
+                username: req.body.username
+            },
+        });
+
+        if (!userData) {
+            res.status(400).json({ message: "Incorrect username or password.  Please try again!" });
+            return;
+        };
+
+        const validPassword = await userData.checkPassword(req.body.password);
+
+        if (!validPassword) {
+            res.status(400).json({ message: "Incorrect username or password. Please try again!" });
+            return;
+        };
+
+        req.session.save(() => {
+            req.session.loggedIn = true;
+            req.session.username = req.body.username;
```

Comment:
Should we also save the user ID so we can use it to show the trips for the user?